libraries/libstratosphere/source/settings/impl/settings_key_value_store.cpp

```diff
@@ -0,0 +1,1629 @@
+                data->SetMountName(FwdbgSystemSaveDataMountName);
+
+                /* Note that we constructed. */
+                s_initialized = true;
+            }
+
+            /* Mount the system data, if we haven't already. */
+            if (AMS_UNLIKELY(!s_mounted)) {
+                /* Mount the system data. */
+                R_TRY(data->Mount(create_save));
+
+                /* Note that we mounted. */
+                s_mounted = true;
+            }
+
+            /* Set the output pointer. */
+            *out_data = data;
+            return ResultSuccess();
+        }
+
+        Result LoadKeyValueStoreMap(Map *out) {
+            /* Check pre-conditions. */
+            AMS_ASSERT(out != nullptr);
+
+            /* Clear the key value store map. */
+            ClearKeyValueStoreMap(*out);
+
+            /* Get the firmware debug system data. */
+            SystemData *system_data = nullptr;
+            R_TRY(GetSystemData<SystemDataTag::Fwdbg>(std::addressof(system_data), ncm::SystemDataId::FirmwareDebugSettings));
+            AMS_ASSERT(system_data != nullptr);
+            
+            /* Load the default keys/values for the firmware debug system data. */
+            R_TRY(LoadKeyValueStoreMapDefault<SystemData>(out, *system_data));
+
+            /* Load the keys/values based on the hardware type. */
+            R_TRY(LoadKeyValueStoreMap(out, GetSplHardwareType()));
+
+            if (IsSplDevelopment()) {
+                /* Get the system save data. */
+                SystemSaveData *system_save_data = nullptr;
+                R_TRY(GetSystemSaveData(std::addressof(system_save_data), false));
+                AMS_ASSERT(system_save_data != nullptr);
+
+                /* Attempt to load the current keys/values from the system save data. */
+                if (const auto result = LoadKeyValueStoreMapCurrent<SystemSaveData>(out, *system_save_data); R_FAILED(result)) {
+                    /* Reset all values to their defaults. */
+                    for (auto it = out->begin(); it != out->end(); ++it) {
+                        MapValue &map_value = it->second;
+
+                        /* Free the current value. */
+                        if (map_value.current_value != nullptr && map_value.current_value != map_value.default_value) {
+                            FreeToHeap(map_value.current_value, map_value.current_value_size);
+                        }
+
+                        /* Reset the current value to the default value. */
+                        map_value.current_value_size = map_value.default_value_size;
+                        map_value.current_value      = map_value.default_value;
+                    }
+
+                    /* Log failure to load system save data. */
+                    AMS_LOG("[firmware debug settings] Warning: Failed to load the system save data. (%08x, %d%03d-%04d)\n", result.GetInnerValue(), 2, result.GetModule(), result.GetDescription());
```

Comment:
This needs to be a warning, not a normal log, but we don't really have infra for that. Please add a TODO comment to make it a warning.